<!DOCTYPE html>
<html>
	<head>
		<meta charset="utf-8">
		<title></title>
	</head>
	<body>
		<script>
			async function fn() {
				// 返回的不是Promise对象，那么这个函数返回的就是一个成功的Promise
				// return 'this is function'
				// return null 也就是undefined
				// return;
				// 抛出错误，则返回结果为一个失败的Promise
				// throw new Error('404')
				// 返回结果为一个Promise 对象
				return new Promise((resolve, reject) => {
					resolve('success')
				})
			}
			const f = fn()

			f.then((value, reason) => {
				console.log(value)
			})
		</script>
	</body>
</html>
